bytesconv_table_gen.go

```diff
@@ -100,23 +100,50 @@ func main() {
 
 	validHeaderFieldByteTable := func() [128]byte {
 		// Should match net/textproto's validHeaderFieldByte(c byte) bool
-		// Defined by RFC 9110 5.6.2
-		//      tchar = "!" / "#" / "$" / "%" / "&" / "'" / "*" / "+" / "-" / "." /
-		//      "^" / "_" / "`" / "|" / "~" / DIGIT / ALPHA
-		var a [128]byte
-		for _, v := range "!#$%&'*+-.^_`|~" {
-			a[v] = 1
-		}
-		for i := 'a'; i <= 'z'; i++ {
-			a[i] = 1
-		}
-		for i := 'A'; i <= 'Z'; i++ {
-			a[i] = 1
+		// Defined by RFC 7230:
+		//
+		//	header-field   = field-name ":" OWS field-value OWS
+		//	field-name     = token
+		//	tchar = "!" / "#" / "$" / "%" / "&" / "'" / "*" / "+" / "-" / "." /
+		//	        "^" / "_" / "`" / "|" / "~" / DIGIT / ALPHA
+		//	token = 1*tchar
+		var table [128]byte
+		for c := 0; c < 128; c++ {
+			if (c >= '0' && c <= '9') ||
+				(c >= 'a' && c <= 'z') ||
+				(c >= 'A' && c <= 'Z') ||
+				c == '!' || c == '#' || c == '$' || c == '%' || c == '&' ||
+				c == '\'' || c == '*' || c == '+' || c == '-' || c == '.' ||
+				c == '^' || c == '_' || c == '`' || c == '|' || c == '~' {
+				table[c] = 1
+			}
 		}
-		for i := '0'; i <= '9'; i++ {
-			a[i] = 1
+		return table
+	}()
+
+	validHeaderValueByteTable := func() [256]byte {
+		// Should match net/textproto's validHeaderValueByte(c byte) bool
+		// Defined by RFC 7230:
+		//
+		//	field-content  = field-vchar [ 1*( SP / HTAB ) field-vchar ]
+		//	field-vchar    = VCHAR / obs-text
+		//	obs-text       = %x80-FF
+		//
+		// RFC 5234:
+		//
+		//	HTAB           =  %x09
+		//	SP             =  %x20
+		//	VCHAR          =  %x21-7E
+		var table [256]byte
+		for c := 0; c < 256; c++ {
+			if (c >= 0x21 && c <= 0x7E) || // VCHAR
+				c == 0x20 || // SP
+				c == 0x09 || // HTAB
+				c >= 0x80 { // obs-text
```

Comment:
RFC 9110 does allow for you to be a little less strict here, if you'd like:
> Field values containing CR, LF, or NUL characters are invalid and dangerous, due to the varying ways that implementations might parse and interpret those characters; a recipient of CR, LF, or NUL within a field value MUST either reject the message or replace each of those characters with SP before further processing or forwarding of that message. Field values containing other CTL characters are also invalid; however, recipients MAY retain such characters for the sake of robustness when they appear within a safe context (e.g., an application-specific quoted string that will not be processed by any downstream HTTP parser).

Reply:
I think it's probably better to just reject these values like `net/http` does. Parsing the value to see if it's in a safe context will be complicated.